src/zeta/aca_oam_server.cpp

```diff
@@ -0,0 +1,321 @@
+  match.vni = get_vpc_id(msg_data.vni);
+
+  return match;
+}
+
+oam_action ACA_Oam_Server::get_oam_action_field(oam_message *oammsg)
+{
+  oam_action action;
+
+  flow_inject_msg msg_data = oammsg->data.msg_inject_flow;
+
+  action.inst_nw_dst = inet_ntoa(msg_data.inst_dst_ip);
+  action.node_nw_dst = inet_ntoa(msg_data.node_dst_ip);
+  action.inst_dl_dst = _get_mac_addr(msg_data.inst_dst_mac);
+  _standardize_mac_address(action.inst_dl_dst);
+  action.node_dl_dst = _get_mac_addr(msg_data.node_dst_mac);
+  _standardize_mac_address(action.node_dl_dst);
+  action.idle_timeout = to_string(msg_data.idle_timeout);
+
+  return action;
+}
+
+void ACA_Oam_Server::_parse_oam_flow_injection(oam_message *oammsg)
+{
+  unsigned long not_care_culminative_time;
+  int overall_rc = EXIT_SUCCESS;
+
+  oam_match match = get_oam_match_field(oammsg);
+  oam_action action = get_oam_action_field(oammsg);
+
+  string remote_host_ip = action.node_nw_dst;
+  string vpc_id = match.vni;
+
+  alcor::schema::NetworkType network_type = alcor::schema::NetworkType::VXLAN;
+
+  if (!aca_is_port_on_same_host(remote_host_ip)) {
+    ACA_LOG_INFO("%s", "port_neighbor not exist!\n");
+    aca_ovs_l2_programmer::ACA_OVS_L2_Programmer::get_instance().create_update_neighbor_port(
+            vpc_id, network_type, remote_host_ip, (uint)stoi(vpc_id), not_care_culminative_time);
+  }
+  overall_rc = aca_oam_server::ACA_Oam_Server::add_direct_path(match, action);
+
+  if (overall_rc == EXIT_SUCCESS) {
+    ACA_LOG_INFO("%s", "Command succeeded!\n");
+  } else {
+    ACA_LOG_ERROR("Command failed!!! overrall_rc: %d\n", overall_rc);
+  }
+
+  return;
+}
+
+void ACA_Oam_Server::_parse_oam_flow_deletion(oam_message *oammsg)
+{
+  int overall_rc = EXIT_SUCCESS;
+  oam_match match = get_oam_match_field(oammsg);
+
+  overall_rc = del_direct_path(match);
+
+  if (overall_rc == EXIT_SUCCESS) {
+    ACA_LOG_INFO("%s", "Command succeeded!\n");
+  } else {
+    ACA_LOG_ERROR("Command failed!!! overrall_rc: %d\n", overall_rc);
+  }
+
+  return;
+}
+
+int ACA_Oam_Server::add_direct_path(oam_match match, oam_action action)
+{
+  int overall_rc = EXIT_SUCCESS;
+
+  string vlan_id = to_string(aca_vlan_manager::ACA_Vlan_Manager::get_instance().get_or_create_vlan_id(
+          match.vni));
+  string outport_name =
+          aca_get_outport_name(alcor::schema::NetworkType::VXLAN, action.node_nw_dst);
+
+  string cmd_match = "ip,nw_proto=" + match.proto + ",nw_src=" + match.sip +
+                     ",nw_dst=" + match.dip + ",tp_src=" + match.sport +
+                     ",tp_dst=" + match.dport + ",dl_vlan=" + vlan_id;
+  string cmd_action = "action=\"strip_vlan,load:" + match.vni +
+                      "->NXM_NX_TUN_ID[],mod_dl_dst=" + action.inst_dl_dst +
+                      ",mod_nw_dst=" + action.inst_nw_dst +
+                      ",idle_timeout=" + action.idle_timeout + ",output:" + outport_name;
+
+  string opt = "table=55,priority=50," + cmd_match + "," + cmd_action;
```

Comment:
using table 55 is for the on-demand routing rule which is not implemented yet. For now, please populate Table 20 unicast (to be added by you :)) and Table 22 multicast